kt/godot-runtime/src/main/kotlin/godot/core/VariantArray.kt

```diff
@@ -1,57 +1,472 @@
+    }
+
+    override fun toString(): String {
+        return "Array(${size})"
+    }
+
+
+    @Suppress("FunctionName")
+    private object Bridge {
+        external fun engine_call_constructor(): VoidPtr
+
+        external fun engine_call_get_size(_handle: VoidPtr)
+        external fun engine_call_clear(_handle: VoidPtr)
+        external fun engine_call_empty(_handle: VoidPtr)
+        external fun engine_call_hash(_handle: VoidPtr)
+        external fun engine_call_invert(_handle: VoidPtr)
+        external fun engine_call_remove(_handle: VoidPtr)
+        external fun engine_call_resize(_handle: VoidPtr)
+        external fun engine_call_shuffle(_handle: VoidPtr)
+        external fun engine_call_sort(_handle: VoidPtr)
+        external fun engine_call_sortCustom(_handle: VoidPtr)
+        external fun engine_call_append(_handle: VoidPtr)
+        external fun engine_call_bsearch(_handle: VoidPtr)
+        external fun engine_call_bsearchCustom(_handle: VoidPtr)
+        external fun engine_call_count(_handle: VoidPtr)
+        external fun engine_call_duplicate(_handle: VoidPtr)
+        external fun engine_call_erase(_handle: VoidPtr)
+        external fun engine_call_find(_handle: VoidPtr)
+        external fun engine_call_findLast(_handle: VoidPtr)
+        external fun engine_call_front(_handle: VoidPtr)
+        external fun engine_call_has(_handle: VoidPtr)
+        external fun engine_call_insert(_handle: VoidPtr)
+        external fun engine_call_max(_handle: VoidPtr)
+        external fun engine_call_min(_handle: VoidPtr)
+        external fun engine_call_popBack(_handle: VoidPtr)
+        external fun engine_call_popFront(_handle: VoidPtr)
+        external fun engine_call_pushBack(_handle: VoidPtr)
+        external fun engine_call_pushFront(_handle: VoidPtr)
+        external fun engine_call_rfind(_handle: VoidPtr)
+        external fun engine_call_slice(_handle: VoidPtr)
+        external fun engine_call_operator_set(_handle: VoidPtr)
+        external fun engine_call_operator_get(_handle: VoidPtr)
     }
 }
 
-inline fun <reified T> VariantArray() =
-        VariantArray<T>(
-                variantMapper[T::class]
-                ?: throw UnsupportedOperationException("Can't create a VariantArray with generic ${T::class}.")
-        )
 
-fun variantArrayOf(vararg args: Any?) = VariantArray<Any?>()
\ No newline at end of file
+//CONSTRUCTOR
+@Suppress("FunctionName")
+inline fun <reified T> VariantArray(): VariantArray<T> {
+    val variantType = variantMapper[T::class]
+    checkNotNull(variantType) {
+        "Can't create a VariantArray with generic ${T::class}."
+    }
+    return VariantArray<T>(
+            variantType
+    )
+}
+
+//HELPER
+inline fun <reified T> variantArrayOf(vararg args: T) = VariantArray<T>().also { it.addAll(args) }
+
+/**
+ * Convert an iterable into a GodotArray
+ * Warning: Might be slow if the iterable contains a lot of items because GDNative can only append items one by one

```

Comment:
Maybe add a Todo here as this is about to change (not sure if only 4.0 or if it will be backported to 3.2

Reply:
Actually, it can be changed right now.
GDNative didn't have this function, but now we are in a module so I can get the true equal operator from the C++

Reply:
We will backport to 3.2 I think, 4.0 should not be so much changes.